Flaky TestGenerateProgram/Azure_Native - sporadic panic

## Hello!
<!-- Please leave this section as-is, it's designed to help others in the community know how to interact with our GitHub issues. -->

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

We a have non-determinstic program gen issue.

### Steps to reproduce

Repeatedly run `go test -run TestGenerateProgram/Azure_Native` until it panics

Expected: <!-- What did you expect to happen? -->
Actual: <!-- What happens instead? -->

I have tried capturing the code and precise errors that are panicking under gofmt:

```/tmp/panic.go:29:1: expected expression```

```go
/tmp/panic.go:29:1: expected expression
package main

import (

"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
cdn "github.com/pulumi/pulumi-azure-native/sdk/go/azure/cdn"
network "github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
)
func main() {
pulumi.Run(func(ctx *pulumi.Context) error {
_, err := network.NewFrontDoor(ctx, "frontDoor", &network.FrontDoorArgs{
RoutingRules: network.RoutingRuleArray{
&network.RoutingRuleArgs{
RouteConfiguration: network.ForwardingConfiguration{
OdataType: "#Microsoft.Azure.FrontDoor.Models.FrontdoorForwardingConfiguration",
BackendPool: network.SubResource{
Id: "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/frontDoors/frontDoor1/backendPools/backendPool1",
},
},
},
},
})
if err != nil {
return err
}
_, err = cdn.NewEndpoint(ctx, "endpoint", &cdn.EndpointArgs{
DeliveryPolicy: &cdn.EndpointPropertiesUpdateParametersDeliveryPolicyArgs{
Rules: []cdn.DeliveryRuleArgs{
interface{}{
Actions: pulumi.AnyArray{
cdn.DeliveryRuleCacheExpirationAction{
Name: "CacheExpiration",
Parameters: cdn.CacheExpirationActionParameters{
CacheBehavior: "Override",
CacheDuration: "10:10:09",
CacheType: "All",
OdataType: "#Microsoft.Azure.Cdn.Models.DeliveryRuleCacheExpirationActionParameters",
},
},
cdn.DeliveryRuleResponseHeaderAction{
Name: "ModifyResponseHeader",
Parameters: cdn.HeaderActionParameters{
HeaderAction: "Overwrite",
HeaderName: "Access-Control-Allow-Origin",
OdataType: "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
Value: "*",
},
},
cdn.DeliveryRuleRequestHeaderAction{
Name: "ModifyRequestHeader",
Parameters: cdn.HeaderActionParameters{
HeaderAction: "Overwrite",
HeaderName: "Accept-Encoding",
OdataType: "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
Value: "gzip",
},
},
},
Conditions: pulumi.AnyArray{
cdn.DeliveryRuleRemoteAddressCondition{
Name: "RemoteAddress",
Parameters: cdn.RemoteAddressMatchConditionParameters{
MatchValues: []string{
"192.168.1.0/24",
"10.0.0.0/24",
},
NegateCondition: true,
OdataType: "#Microsoft.Azure.Cdn.Models.DeliveryRuleRemoteAddressConditionParameters",
Operator: "IPMatch",
},
},
},
Name: pulumi.String("rule1"),
Order: pulumi.Int(1),
},
},
},
EndpointName: pulumi.String("endpoint1"),
IsCompressionEnabled: pulumi.Bool(true),
IsHttpAllowed: pulumi.Bool(true),
IsHttpsAllowed: pulumi.Bool(true),
Location: pulumi.String("WestUs"),
ProfileName: pulumi.String("profileName"),
ResourceGroupName: pulumi.String("resourceGroupName"),
})
if err != nil {
return err
}
return nil
})
}

```

--- a/pkg/codegen/testing/test/program_driver.go
+++ b/pkg/codegen/testing/test/program_driver.go
@@ -137,8 +137,6 @@ var PulumiPulumiProgramTests = []ProgramTest{
 	{
 		Directory:   "azure-native",
 		Description: "Azure Native",
-		Skip:        codegen.NewStringSet("go"),
-		// Blocked on TODO[pulumi/pulumi#8123]
 		SkipCompile: codegen.NewStringSet("go", "nodejs", "dotnet"),
 		// Blocked on go:
 		//   TODO[pulumi/pulumi#8072]

--- a/pkg/codegen/testing/test/testdata/azure-native-pp/go/azure-native.go
+++ b/pkg/codegen/testing/test/testdata/azure-native-pp/go/azure-native.go
@@ -1,14 +1,15 @@
 package main
 
 import (
-	cdn "github.com/pulumi/pulumi-azure-native/sdk/go/azure/cdn"
-	network "github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/cdn"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
 func main() {
 	pulumi.Run(func(ctx *pulumi.Context) error {
 		_, err := network.NewFrontDoor(ctx, "frontDoor", &network.FrontDoorArgs{
+			ResourceGroupName: pulumi.String("someGroupName"),
 			RoutingRules: network.RoutingRuleArray{
 				&network.RoutingRuleArgs{
 					RouteConfiguration: network.ForwardingConfiguration{
@@ -24,31 +25,32 @@ func main() {
 			return err
 		}
 		_, err = cdn.NewEndpoint(ctx, "endpoint", &cdn.EndpointArgs{
+			Origins: cdn.DeepCreatedOriginArray{},
 			DeliveryPolicy: &cdn.EndpointPropertiesUpdateParametersDeliveryPolicyArgs{
 				Rules: []cdn.DeliveryRuleArgs{
-					&cdn.DeliveryRuleArgs{
+					{
 						Actions: pulumi.AnyArray{
-							cdn.DeliveryRuleCacheExpirationAction{
+							{
 								Name: "CacheExpiration",
-								Parameters: cdn.CacheExpirationActionParameters{
+								Parameters: {
 									CacheBehavior: "Override",
 									CacheDuration: "10:10:09",
 									CacheType:     "All",
 									OdataType:     "#Microsoft.Azure.Cdn.Models.DeliveryRuleCacheExpirationActionParameters",
 								},
 							},
-							cdn.DeliveryRuleResponseHeaderAction{
+							{
 								Name: "ModifyResponseHeader",
-								Parameters: cdn.HeaderActionParameters{
+								Parameters: {
 									HeaderAction: "Overwrite",
 									HeaderName:   "Access-Control-Allow-Origin",
 									OdataType:    "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
 									Value:        "*",
 								},
 							},
-							cdn.DeliveryRuleRequestHeaderAction{
+							{
 								Name: "ModifyRequestHeader",
-								Parameters: cdn.HeaderActionParameters{
+								Parameters: {
 									HeaderAction: "Overwrite",
 									HeaderName:   "Accept-Encoding",
 									OdataType:    "#Microsoft.Azure.Cdn.Models.DeliveryRuleHeaderActionParameters",
@@ -57,9 +59,9 @@ func main() {
 							},
 						},
 						Conditions: pulumi.AnyArray{
-							cdn.DeliveryRuleRemoteAddressCondition{
+							{
 								Name: "RemoteAddress",
-								Parameters: cdn.RemoteAddressMatchConditionParameters{
+								Parameters: {
 									MatchValues: []string{
 										"192.168.1.0/24",
 										"10.0.0.0/24",

